### ISSUE TITLE
[BUG] SleekXMPP should be replaced with slixmpp

### ISSUE THREAD

**Issue by ari** [saltstack/salt#60354]
The slixmpp project has replaced the now deprecated SleekXMPP.  See the notice here: https://github.com/fritzy/SleekXMPP

On the plus side, the new library should be just drop in with only import changes.

**Comment by sagetherage** [comment]
@cmcmarrow let's put in a deprecation warning in and then get an estimate of LOE level of effort for the new library. Warning can go in right away, but need to get an idea of LOE before we can implement the replacement and plan for it in a release. Let's do the planning in `Silicon` release cycle.

**Comment by ari** [comment]
Is it clear that anything user facing needs to be deprecated, other than ensuring the user puts the correct library into the python classpath?

**Comment by cmcmarrow** [comment]
@sagetherage slixmpp API looks to be the same "took a 30 sec look". If that is true we should try to import it and then import the older module, as well add the deprecation warning.

**Comment by cmcmarrow** [comment]
https://github.com/fritzy/SleekXMPP repository has been archived AKA it is no long maintained.

**Comment by sagetherage** [comment]
@cmcmarrow yes - level of effort is needed, given all the other priorities for the current release cycle
